langchain4j-azure-cosmos-no-sql/src/test/java/dev/langchain4j/store/embedding/azure/cosmos/no/sql/AzureCosmosDbNoSqlEmbeddingStoreIT.java

```diff
@@ -0,0 +1,189 @@
+import com.azure.cosmos.CosmosDatabase;
+import com.azure.cosmos.models.CosmosContainerProperties;
+import com.azure.cosmos.models.CosmosVectorDataType;
+import com.azure.cosmos.models.CosmosVectorDistanceFunction;
+import com.azure.cosmos.models.CosmosVectorEmbedding;
+import com.azure.cosmos.models.CosmosVectorEmbeddingPolicy;
+import com.azure.cosmos.models.CosmosVectorIndexSpec;
+import com.azure.cosmos.models.CosmosVectorIndexType;
+import com.azure.cosmos.models.IncludedPath;
+import com.azure.cosmos.models.IndexingMode;
+import com.azure.cosmos.models.IndexingPolicy;
+import com.azure.cosmos.models.PartitionKeyDefinition;
+import dev.langchain4j.data.embedding.Embedding;
+import dev.langchain4j.data.segment.TextSegment;
+import dev.langchain4j.model.embedding.AllMiniLmL6V2QuantizedEmbeddingModel;
+import dev.langchain4j.model.embedding.EmbeddingModel;
+import dev.langchain4j.store.embedding.EmbeddingMatch;
+import dev.langchain4j.store.embedding.EmbeddingStore;
+import dev.langchain4j.store.embedding.EmbeddingStoreIT;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.EnabledIfEnvironmentVariable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import static java.util.Arrays.asList;
+import static org.assertj.core.api.Assertions.assertThat;
+
+@EnabledIfEnvironmentVariable(named = "AZURE_COSMOS_HOST", matches = ".+")
+@EnabledIfEnvironmentVariable(named = "AZURE_COSMOS_MASTER_KEY", matches = ".+")
+public class AzureCosmosDbNoSqlEmbeddingStoreIT extends EmbeddingStoreIT {
+
+    protected static Logger logger = LoggerFactory.getLogger(AzureCosmosDbNoSqlEmbeddingStoreIT.class);
+
+    private static final String DATABASE_NAME = "test_database_langchain_java";
+    private static final String CONTAINER_NAME = "test_container";
+    private CosmosClient client;
+    CosmosDatabase database;
+    private final EmbeddingModel embeddingModel;
+    private final EmbeddingStore<TextSegment> embeddingStore;
+    private final int dimensions;
+    private final String HOST = System.getenv("AZURE_COSMOS_HOST");
+    private final String KEY = System.getenv("AZURE_COSMOS_MASTER_KEY");
+
+    public AzureCosmosDbNoSqlEmbeddingStoreIT() {
+        embeddingModel = new AllMiniLmL6V2QuantizedEmbeddingModel();
+        dimensions = embeddingModel.embed("hello").content().vector().length;
+
+        client = new CosmosClientBuilder()
+                .endpoint(HOST)
+                .key(KEY)
+                .consistencyLevel(ConsistencyLevel.EVENTUAL)
+                .contentResponseOnWriteEnabled(true)
+                .buildClient();
+
+        embeddingStore = AzureCosmosDbNoSqlEmbeddingStore.builder()
+                .cosmosClient(client)
+                .databaseName(DATABASE_NAME)
+                .containerName(CONTAINER_NAME)
+                .cosmosVectorEmbeddingPolicy(populateVectorEmbeddingPolicy(dimensions))
```

Comment:
please double check which properties are needed only for creating database/container if it does not exists and which are needed for querying. I suspect some of these properties are not required once db/container are created

Reply:
@langchain4j All these properties are required to create a container/database which can perform vector search. Only the path of the embedding path is required for querying. 

I'm not sure what else do you want to me check here.

Reply:
Can you implement it so that user does not have to provide **_all_** properties (see all the assertions for null/empty in the constructor) if the database/container are already in place?